sympy/geometry/entity.py

```diff
@@ -537,9 +537,9 @@ def _contains(self, other):
         """sympy.sets uses the _contains method, so include it for compatibility."""
 
         if isinstance(other, Set) and other.is_FiniteSet:
-            return all(self.__contains__(i) for i in other)
+            return all(i in self for i in other)
```

Comment:
This could be
```
return And(*(self._contains(i) for i in other))
```
except that this should return `True/False` rather than `S.true/S.false` if the condition evaluates.

Reply:
This class is a mess, but since all the subclasses implement only `__contains__`  returning a bool, I don't think this makes sense.

Reply:
`__contains__` is only allowed to return a boolean in Python.

Reply:
As subclasses of `Set` all the `GeometrySet` subclasses should implement `_contains` rather than `__contains__`:
https://github.com/sympy/sympy/blob/7437631ba09b240c0647129bde8896579c17ad58/sympy/sets/sets.py#L668-L676

Reply:
`GeometrySet` inherits `__contains__` from `GeometryEntity`, its first base class:

https://github.com/sympy/sympy/blob/7437631ba09b240c0647129bde8896579c17ad58/sympy/geometry/entity.py#L100-L104

not from `Set`

Reply:
This is a bit messy. Plane and Point subclass GeometryEntity which itself subclasses Basic. Ellipse, LinearEntity, Parabola and Polygon all subclass GeometrySet which is a subclass of both GeometryEntity and Set.

I think that the intention is that all geometry objects should behave like sets of points. Maybe it would be a cleaner to have methods for converting them into a particular representation of sets of points rather than literally making all of these objects sets.

Reply:
Related to this issue https://github.com/sympy/sympy/issues/11683. I agree that geometric entities should not be Set objects. Geometric operations are similar to general set operations, but I think they aren't exactly the same. For example, it only makes sense for a geometric "set" to contain geometric objects, whereas Set can hold anything.